src/components/WorkPoolStatusIcon.vue

```diff
@@ -15,17 +15,22 @@
 </template>
 
 <script lang="ts" setup>
-  import { computed } from 'vue'
+  import { computed, ref } from 'vue'
   import StatusIcon from '@/components/StatusIcon.vue'
-  import { WorkPoolStatus } from '@/models'
+  import { WorkPool } from '@/models'
 
   const props = defineProps<{
-    status: WorkPoolStatus,
+    workPool: WorkPool,
   }>()
 
+  const status = ref(props.workPool.status)
```

Comment:
You'll want to make this a computed otherwise you'll break reactivity. (Remove the `ref` import as well as it will be unused)
```suggestion
  const status = computed(() => props.workPool.status)
```